test/test_model.py

```diff
@@ -434,7 +434,7 @@ def test_pod_spec(self):
         model = ops.model.Model(meta, ops.model._ModelBackend('myapp/0'))
         fake_script(self, 'pod-spec-set', """
                     cat $2 > $(dirname $0)/spec.json
-                    [[ -n $4 ]] && cat $4 > $(dirname $0)/k8s_res.json || true
+                    [ -n "$4" ] && cat $4 > $(dirname $0)/k8s_res.json || true
```

Comment:
if quotes are needed for $4 in the first case, why not in the second one? (for `cat`)

Reply:
the quotes are needed so that test (aka [ ]) is guaranteed an argument. Once you have done '-n' you know that $4 is not empty, so the cat is guaranteed to have an argument.
That said, I don't mind having the extra "", but there is reason why it isn't *required*.

Reply:
as to why they're needed in the first place, `[[ ... ]]` is a bashism that is smart enough to evaluate `[[ -n $EMPTY_VAR ]]` as `[[ -n "" ]]` which evaluates to false (the empty string is not empty), but `[` is not: `[ -n $EMPTY_VAR ]` is the same as `[ -n ]` which is true (!!!).